Unable to create XsltPayloadTransformer when Saxon HE is in the classpath.

<!--
!!! To report Security Vulnerabilities, please go to https://pivotal.io/security !!!
-->
**Affects Version(s):** 4.3.19.RELEASE

---
<!--
Thanks for raising a Spring Integration issue. What sort of issue are you raising?

## Question

Please ask questions about how to use something, or to understand why something isn't
working as you expect it to, on
[Stack Overflow using the `spring-integration` tag](https://stackoverflow.com/questions/tagged/spring-integration).

## Bug report

Please provide details of the problem, including the version of Spring Integration that you
are using. If possible, please provide a test case or sample application that reproduces
the problem. This makes it much easier for us to diagnose the problem and to verify that
we have fixed it.

## Enhancement

Please start by describing the problem that you are trying to solve. There may already
be a solution, or there may be a way to solve it that you hadn't considered.
-->

After upgrading to spring-integration-xml:4.3.19.RELEASE (from 4.3.18.RELEASE), my application failed  to start with the following exception :


```
Error creating bean with name 'XsltPayloadTransformer': Invocation of init method failed;

...

Caused by: java.lang.IllegalArgumentException: Unknown configuration property http://javax.xml.XMLConstants/property/accessExternalDTD
```

The cause of the exception seems to be related with the following changes 

```
- transformerFactory = TransformerFactory.newInstance();
+ transformerFactory = TransformerFactoryUtils.newInstance();
```

The TransformerFactoryUtils.newInstance() force the XMLConstants.ACCESS_EXTERNAL_DTD property which doesn't seems to be supported by Saxon. If I quote answers from Michael Kay about similar subject, I understand that Saxon does not handle Xml Parser related properties.

> Saxon doesn't have control of the XML parsing: we invoke the parser that the application supplies and configures.

As the problem is related to a third party dependency, I provide a [side sample project](https://github.com/stoussaint/sixml-saxon-regression) as a regression test case

--- a/spring-integration-xml/src/main/java/org/springframework/integration/xml/transformer/XsltPayloadTransformer.java
+++ b/spring-integration-xml/src/main/java/org/springframework/integration/xml/transformer/XsltPayloadTransformer.java
@@ -49,6 +49,7 @@ import org.springframework.integration.xml.source.SourceFactory;
 import org.springframework.messaging.Message;
 import org.springframework.messaging.MessagingException;
 import org.springframework.util.Assert;
+import org.springframework.util.ClassUtils;
 import org.springframework.util.ObjectUtils;
 import org.springframework.util.PatternMatchUtils;
 import org.springframework.util.StringUtils;
@@ -225,22 +226,37 @@ public class XsltPayloadTransformer extends AbstractXmlTransformer implements Be
 		super.onInit();
 		this.evaluationContext = ExpressionUtils.createStandardEvaluationContext(getBeanFactory());
 		if (this.templates == null) {
-			TransformerFactory transformerFactory;
-			if (this.transformerFactoryClassName != null) {
-				transformerFactory = TransformerFactory.newInstance(this.transformerFactoryClassName, this.classLoader);
-			}
-			else {
-				transformerFactory = TransformerFactoryUtils.newInstance();
-			}
-			transformerFactory.setAttribute(XMLConstants.ACCESS_EXTERNAL_DTD, "");
-			transformerFactory.setAttribute(XMLConstants.ACCESS_EXTERNAL_STYLESHEET, "file,jar:file");
 			try {
+			TransformerFactory transformerFactory = createTransformerFactory();
 				this.templates = transformerFactory.newTemplates(createStreamSourceOnResource(this.xslResource));
 			}
-			catch (TransformerConfigurationException | IOException e) {
+			catch (ClassNotFoundException | TransformerConfigurationException | IOException e) {
 				throw new IllegalStateException(e);
 			}
 		}
+	}
+
+	private TransformerFactory createTransformerFactory() throws ClassNotFoundException {
+		TransformerFactory transformerFactory;
+		if (this.transformerFactoryClassName != null) {
+			@SuppressWarnings("unchecked")
+			Class<TransformerFactory> transformerFactoryClass =
+					(Class<TransformerFactory>) ClassUtils.forName(this.transformerFactoryClassName, this.classLoader);
+			transformerFactory = TransformerFactoryUtils.newInstance(transformerFactoryClass);
+		}
+		else {
+			transformerFactory = TransformerFactoryUtils.newInstance();
+		}
+		try {
+			transformerFactory.setAttribute(XMLConstants.ACCESS_EXTERNAL_STYLESHEET, "file,jar:file");
+		}
+		catch (IllegalArgumentException ex) {
+			if (logger.isInfoEnabled()) {
+				logger.info("The '" + XMLConstants.ACCESS_EXTERNAL_STYLESHEET + "' property is not supported by "
+						+ transformerFactory.getClass().getCanonicalName());
+			}
+		}
+		return transformerFactory;
 	}
 
 	@Override

--- a/spring-integration-xml/src/test/java/org/springframework/integration/xml/transformer/XsltPayloadTransformerTests.java
+++ b/spring-integration-xml/src/test/java/org/springframework/integration/xml/transformer/XsltPayloadTransformerTests.java
@@ -26,7 +26,6 @@ import javax.xml.transform.Result;
 import javax.xml.transform.Templates;
 import javax.xml.transform.TransformerException;
 import javax.xml.transform.TransformerFactory;
-import javax.xml.transform.TransformerFactoryConfigurationError;
 import javax.xml.transform.dom.DOMResult;
 
 import org.junit.Before;
@@ -161,10 +160,12 @@ public class XsltPayloadTransformerTests {
 
 	@Test
 	public void testXsltPayloadWithBadTransformerFactoryClassName() throws IOException {
-		XsltPayloadTransformer transformer = new XsltPayloadTransformer(getXslResourceThatOutputsText(), "foo.bar.Baz");
+		XsltPayloadTransformer transformer =
+				new XsltPayloadTransformer(getXslResourceThatOutputsText(), "foo.bar.Baz");
 		transformer.setBeanFactory(Mockito.mock(BeanFactory.class));
 		assertThatThrownBy(transformer::afterPropertiesSet)
-				.isExactlyInstanceOf(TransformerFactoryConfigurationError.class);
+				.isExactlyInstanceOf(IllegalStateException.class)
+				.hasCauseExactlyInstanceOf(ClassNotFoundException.class);
 	}
 
 	@Test

